elements/pfelement/src/pfelement.js

```diff
@@ -139,6 +139,15 @@ class PFElement extends HTMLElement {
       });
     }
 
+    // Throw a warning if the on attribute was manually added
+    if (this.hasAttribute("on")) {
```

Comment:
Can we move this out of the constructor and into the connected callback? The spec has this line about things you shouldn't do in the constructor.

"The element's attributes and children must not be inspected"
https://html.spec.whatwg.org/multipage/custom-elements.html#custom-element-conformance

Reply:
Also, when we move this to the connectedCallback(), if the element is ever removed and added somewhere else on the page, I think the connectedCallback() fires again. So if a component gets the `on` attributed after upgrade, and is moved and added somewhere else on the page, this warning will fire. We'd need to test that. A simple check for `this.hasAttribute("pfelement")` after the ShadyCSS check and before the `pfelement` attribute is added could be the right place for the warning.

```
this.connected = true;
this.log(`Connecting...`);

if (window.ShadyCSS) {
  this.log(`Styling...`);
  window.ShadyCSS.styleElement(this);
  this.log(`Styled.`);
}

// PUT THE CHECK HERE
if (!this.hasAttribute("pfelement")) {
  if (this.hasAttribute("on")) {
    ...
  }
}

// @TODO maybe we should use just the attribute instead of the class?
// https://github.com/angular/angular/issues/15399#issuecomment-318785677
this.classList.add("PFElement");
this.setAttribute("pfelement", "");
```

Reply:
I will try it with the hasAttribute, I put it in the connectedCallback originally but it threw the error on every component because it would run it again after instantiation.

Reply:
Perfect!  That did the trick!  Thank you!